### PR TITLE
chore: use the proper http method for joining threads and remove unused methods

### PR DESCRIPTION
**Pull request by onerandomusername**
## Summary

https://discord.com/developers/docs/resources/channel#join-thread

> `PUT` `/channels/{channel.id}/thread-members/@me`

Additionally, removes an unused method from http.py, http.logout, while we're here. This endpoint can not be used by bots.

## Checklist

- [x] If code changes were made, then they have been tested
    - [ ] I have updated the documentation to reflect the changes
    - [x] I have formatted the code properly by running `task lint` or `pre-commit run --all-files`
- [x] This PR fixes an issue
- [ ] This PR adds something new (e.g. new method or parameters)
- [ ] This PR is a breaking change (e.g. methods or parameters removed/renamed)
- [ ] This PR is **not** a code change (e.g. documentation, README, ...)

### CODE REVIEW

**Comment by onerandomusername** [comment]
Increased scope slightly to remove unused method from http while we're here.

File: disnake/http.py [hunk 1]
```diff
@@ -933,6 +917,16 @@ def edit_voice_state(
         )
         return self.request(r, json=payload)
 
+    def edit_my_member(
+        self,
+        guild_id: Snowflake,
+        *,
+        reason: Optional[str] = None,
+        **fields: Any,
```
Comment (shiftinv) [inline disnake/http.py:925]: Any reason not to explicitly specify allowed parameters here? The method returns a `member.Nickname` dict, I feel like either the parameters should be specified or the return type should be made more generic

Comment (onerandomusername) [reply disnake/http.py:925]: I based the method definition here off of `edit_member` (below), which uses `**fields: Any` with no specific validation.

Comment (onerandomusername) [reply disnake/http.py:925]: 7bba3032c873514542727845f9815305b91b1b8a

**Review by shiftinv — APPROVED** [review]
lgtm 👍

**Pull request merged**